ml/ContinuousGesturePredictor.py

```diff
@@ -196,7 +196,8 @@ def showStatistics(predictedClass, confidence):
 convnet=fully_connected(convnet,1000,activation='relu')
 convnet=dropout(convnet,0.75)
 
-convnet=fully_connected(convnet,3,activation='softmax')
+# second parameter in fully_connected is number of classes
+convnet=fully_connected(convnet,4,activation='softmax')
 
 convnet=regression(convnet,optimizer='adam',learning_rate=0.001,loss='categorical_crossentropy',name='regression')
```

Comment:
Have we tried this with different optimizers? Is Adam the best for our case?
Same about learning rate

Reply:
We haven't tried different optimizers. I believe Adam is one of the best optimizers out there according to some [sources](http://ruder.io/optimizing-gradient-descent/) however, [others](https://shaoanlu.wordpress.com/2017/05/29/sgd-all-which-one-is-the-best-optimizer-dogs-vs-cats-toy-experiment/) argue that it might not be as good. We should definitely play around with the parameters. One thing's for sure though, we can probably reduce our epoch from 50 to maybe around ~30 when training our model.